airflow/models/taskinstance.py

```diff
@@ -1839,6 +1839,35 @@ def key(self) -> TaskInstanceKey:
         """Returns a tuple that identifies the task instance uniquely."""
         return TaskInstanceKey(self.dag_id, self.task_id, self.run_id, self.try_number, self.map_index)
 
+    @staticmethod
+    @internal_api_call
+    def _set_state(ti: TaskInstance | TaskInstancePydantic, state, session: Session) -> bool:
+        TI = TaskInstance
+        if not isinstance(ti, TaskInstance):
+            ti = (
+                session.query(TI)
```

Comment:
[Query API](https://docs.sqlalchemy.org/en/20/orm/queryguide/query.html) is deprecated in SQLAlchemy, could we use [session.execute](https://docs.sqlalchemy.org/en/14/orm/session_api.html#sqlalchemy.orm.Session.execute) here ?